src/main/CommandLine.cpp

```diff
@@ -916,11 +939,26 @@ int
 runNewDB(CommandLineArgs const& args)
 {
     CommandLine::ConfigOption configOption;
+    bool resetInMemoryState = false;
 
-    return runWithHelp(args, {configurationParser(configOption)}, [&] {
-        initializeDatabase(configOption.getConfig());
-        return 0;
-    });
+    auto resetParser = [](bool& resetInMemoryState) {
+        return clara::Opt{resetInMemoryState}["--reset-for-in-memory-mode"](
```

Comment:
should use `inMemoryParser`: it was added so that we use the same option across subcommands

Reply:
The call `new-db --in-memory` is confusing and contradicts your other comment about calling the database "an in-memory database". The flag is called "in-memory", however the command sets a persistent state, that the next commands also depend on. I'd like to be as clear to the users as possible.

Reply:
I suggest reusing the term you're using above: minimal. I.e. `stellar-core new-db --minimal-for-in-memory-mode`

Reply:
(I was especially concerned by the word "reset" here because this suggests a non-minimal DB can be changed / reset into a minimal one, and I don't _think_ that's possible?)